src/forwardBeamPipeBrazil.cpp

```diff
@@ -0,0 +1,570 @@
+
+        int pieceIdx = 0; //Larger, rectangular pipe transporting proton and neutral envelopes (neutral exit window and transfer to smaller proton line at the end)
+
+        Box inner_subtract_outer(dd4hep::mm*outerXRadius[5], dd4hep::mm*outerYRadius[5], dd4hep::mm*length[5]/2);
+        Box inner_subtract_inner(dd4hep::mm*innerXRadius[5], dd4hep::mm*innerYRadius[5], dd4hep::mm*(length[5]+5.0)/2);
+
+        Box RP_subtract_outer(dd4hep::mm*outerXRadius[1], dd4hep::mm*outerYRadius[1], dd4hep::mm*(length[2]+5.0)/2);
+        Box RP_subtract_inner(dd4hep::mm*innerXRadius[1], dd4hep::mm*innerYRadius[1], dd4hep::mm*(length[2]+10.0)/2);
+
+        Box pipeAfterB1APF_outer(dd4hep::mm*outerXRadius[pieceIdx], dd4hep::mm*outerYRadius[pieceIdx], dd4hep::mm*length[pieceIdx]/2);
+        Box pipeAfterB1APF_inner(dd4hep::mm*innerXRadius[pieceIdx], dd4hep::mm*innerYRadius[pieceIdx], dd4hep::mm*(length[pieceIdx])/2);
+        Box pipeAfterB1APF_firstEndCap(dd4hep::mm * outerXRadius[pieceIdx], dd4hep::mm * outerYRadius[pieceIdx], dd4hep::mm*5.0/2.0);
+        Tube neutral_exit_window_cutout(0.0, dd4hep::mm*windowRadius, 1.0); // 1.0cm thick
+        Box protonTransferWindow(dd4hep::mm * 155.0, dd4hep::mm*outerYRadius[1], dd4hep::mm*(5.0/2));
+
+        SubtractionSolid tmpAfterB1APF(pipeAfterB1APF_outer, pipeAfterB1APF_inner); //This gets rid of the inner portion of the pipe, but leaves the endcaps
+        SubtractionSolid tmpAfterFrontEndCap(tmpAfterB1APF, pipeAfterB1APF_firstEndCap, Position(0.0, 0.0, dd4hep::mm * (-length[pieceIdx])/2));
+        SubtractionSolid pipeAfterProtonTransferWindow(tmpAfterFrontEndCap, protonTransferWindow, Position(dd4hep::mm * (-120.0), 0.0, dd4hep::mm * (length[pieceIdx])/2 ));
+
+        SubtractionSolid pipeAfterB1APF(pipeAfterProtonTransferWindow, neutral_exit_window_cutout, Position(dd4hep::mm * 160.0 , 0.0, dd4hep::mm * 0.5*length[pieceIdx]));
+
+    Volume v_pipeAfterB1APF(Form("v_pipeAfterB1APF_%d", pieceIdx), pipeAfterB1APF, m_SS);
+    sdet.setAttributes(det, v_pipeAfterB1APF, x_det.regionStr(), x_det.limitsStr(), vis_name);
+
+    auto pv_pipe_0 = assembly.placeVolume(v_pipeAfterB1APF, Transform3D(RotationY(-0.025), Position(dd4hep::mm * xCenter[pieceIdx] + 4.0, dd4hep::mm *  yCenter[pieceIdx], dd4hep::mm * zCenter[pieceIdx]))); // 2353.06094)));
+    pv_pipe_0.addPhysVolID("sector", 1);
+    DetElement pipe_de_0(sdet, Form("sector_pipe_%d_de", pieceIdx), 1);
+    pipe_de_0.setPlacement(pv_pipe_0);
+
+        //--------------------------------------------------------------------
+
+        pieceIdx = 1; //smaller rectangular pipe for the protons
+
+        Box pipe_1_outer(dd4hep::mm*outerXRadius[pieceIdx], dd4hep::mm*outerYRadius[pieceIdx], dd4hep::mm*length[pieceIdx]/2);
+        Box pipe_1_inner(dd4hep::mm*innerXRadius[pieceIdx], dd4hep::mm*innerYRadius[pieceIdx], dd4hep::mm*(length[pieceIdx]+5.0)/2);
+        SubtractionSolid pipe_1(pipe_1_outer, pipe_1_inner);
```

Comment:
All of these subtraction solids should be an inner box placed inside the outer box, not a subtraction solid.

Reply:
Done.